### ISSUE TITLE
Is there a good way to reference/render a diagram which resides in a different location? 

### ISSUE THREAD

**Issue by wolfspyre** [mingrammer/diagrams#487]
I'd like to know if there's a standard / supported / good / way to reference diagrams which reside in another file / repository?

If so, would there be support for backlinking? 
inheritance? 

🤡 . o O ( so many rabbitholes... ) 

My general use case is a diagram which represents a superset several different services' diagrams represented canonically in their respective repositories.. and was thinking that mermaid may have some metapreprocessing available to link between markdown files which contain mermaid diagrams in a more fluid fashion.... 

prolly several ways to accomplish the goal, but thought there might be one that's more elegant, and if so, was wondering if that was a pattern that aught be reinforced?

**Comment by clayms** [comment]
I don't think this is exactly what you are after, but you can use this library to save the dot file, then using the underlying graphviz library load it and render it.  See below.

1. Create a graph with this library and save it as a dot file by calling the chained methods `.dot.save("filename.gv")` on your Diagram.  
1. Then use the `Source` class from the graphviz python library to load the file and render it into your chosen format.    

### 1:

```python
from diagrams import Diagram
from diagrams.k8s.clusterconfig import HPA
from diagrams.k8s.compute import Deployment, Pod, ReplicaSet
from diagrams.k8s.network import Ingress, Service

with Diagram("Exposed Pod with 3 Replicas", show=False) as diag_1:
    net = Ingress("domain.com") >> Service("svc")
    net >> [Pod("pod1"),
            Pod("pod2"),
            Pod("pod3")] << ReplicaSet("rs") << Deployment("dp") << HPA("hpa")


diag_1.dot.save("diag_1.gv)
```

### 2:

```python
from graphviz import Source

diag_import = Source.from_file(filename="diag_1.gv", format="png")

diag_import.render()
```

See https://graphviz.readthedocs.io/en/stable/api.html#source

When you could load the `.gv` file its type is `graphviz.files.Source`.  If I could convert its type to `graphviz.dot.Digraph`, then I could add it to another `Diagram` as a subgraph using the code below.  However, this gives an error as you cannot do that with a `graphviz.files.Source` type, only with a `graphviz.dot.Digraph`.

```python
with Diagram("", show=False, ) as out_diag:
    pass

out_diag.subgraph(diag_import.dot)

out_diag
```

Perhaps you could pickle the initial `Diagram` (`diag_1` ) as a binary and then reload that... I'll try.

**Comment by clayms** [comment]
@wolfspyre   Ok... I figured it out.

### 1. create and save diagram as dot file

```python
from diagrams import Diagram
from diagrams.k8s.clusterconfig import HPA
from diagrams.k8s.compute import Deployment, Pod, ReplicaSet
from diagrams.k8s.network import Ingress, Service

with Diagram("Exposed Pod with 3 Replicas", show=False) as diag_1:
    net = Ingress("domain.com") >> Service("svc")
    net >> [Pod("pod1"),
            Pod("pod2"),
            Pod("pod3")] << ReplicaSet("rs") << Deployment("dp") << HPA("hpa")


diag_1.dot.save("diag_1.gv)
```

### 2.  create `graphviz.dot.Digraph` from dot file

```python
from graphviz import Digraph, Source

diag_import_body = Source.from_file(filename="diag_1.gv", format="png").source.split('\n')[1:-2]
diag_import = Digraph(body=diag_import_body)
```

### 3. Create "blank" `Diagram` and add imported `Digraph` as subgraph

```python

###
with Diagram("", show=False, ) as final_diag:
    pass

final_diag.subgraph(diag_import)

final_diag
```

![image](https://user-images.githubusercontent.com/12793783/111363383-5bced580-8666-11eb-8f2c-3ba4305e8ddd.png)

**Comment by clayms** [comment]
@wolfspyre   Also see https://github.com/mingrammer/diagrams/issues/436#issuecomment-800563909 to see how to add multiple separate graphs to a new `Diagram`
